iceoryx_posh/include/iceoryx_posh/internal/popo/ports/client_port_user.hpp

```diff
@@ -0,0 +1,117 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#ifndef IOX_POSH_POPO_PORTS_CLIENT_PORT_USER_HPP
+#define IOX_POSH_POPO_PORTS_CLIENT_PORT_USER_HPP
+
+#include "iceoryx_posh/internal/popo/building_blocks/chunk_receiver.hpp"
+#include "iceoryx_posh/internal/popo/building_blocks/chunk_sender.hpp"
+#include "iceoryx_posh/internal/popo/ports/base_port.hpp"
+#include "iceoryx_posh/internal/popo/ports/client_port_data.hpp"
+#include "iceoryx_posh/mepoo/chunk_header.hpp"
+#include "iceoryx_utils/cxx/expected.hpp"
+#include "iceoryx_utils/cxx/helplets.hpp"
+#include "iceoryx_utils/cxx/optional.hpp"
+#include "iceoryx_utils/error_handling/error_handling.hpp"
+
+namespace iox
+{
+namespace popo
+{
+/// @brief The ClientPortUser provides the API for accessing a client port from the user side. The client port
+/// is divided in the three parts ClientPortData, ClientPortRouDi and ClientPortUser. The ClientPortUser
+/// uses the functionality of a ChunkSender and ChunReceiver for sending requests and receiving responses.
+/// Additionally it provides the connect / disconnect API which controls whether the client port shall connect to the
+/// server
+class ClientPortUser : public BasePort
+{
+  public:
+    using MemberType_t = ClientPortData;
+
+    explicit ClientPortUser(cxx::not_null<MemberType_t* const> clientPortDataPtr) noexcept;
+
+    ClientPortUser(const ClientPortUser& other) = delete;
+    ClientPortUser& operator=(const ClientPortUser&) = delete;
+    ClientPortUser(ClientPortUser&& rhs) = default;
+    ClientPortUser& operator=(ClientPortUser&& rhs) = default;
+    ~ClientPortUser() = default;
+
+    /// @brief Allocate a chunk, the ownerhip of the SharedChunk remains in the ClientPortUser for being able to
+    /// cleanup if the user process disappears
+    /// @param[in] payloadSize, size of the user paylaod without additional headers
+    /// @return on success pointer to a ChunkHeader which can be used to access the payload and header fields, error if
+    /// not
+    cxx::expected<RequestHeader*, AllocationError> allocateRequest(const uint32_t payloadSize) noexcept;
+
+    /// @brief Free an allocated request without sending it
+    /// @param[in] chunkHeader, pointer to the ChunkHeader to free
+    void freeRequest(RequestHeader* const requestHeader) noexcept;
+
+    /// @brief Send an allocated request chunk to the server port
+    /// @param[in] chunkHeader, pointer to the ChunkHeader to send
+    void sendRequest(RequestHeader* const requestHeader) noexcept;
+
+    /// @brief try to connect to the server
```

Comment:
Since you are mentioning "try to connect to the server" here I would add a small code example on howto use this method correctly. Maybe something like this:
```
/// @code 
///   myPort.connect();
///   while(myPort.getConnectionState() != ConnectionState::CONNECTED) 
///       sleep(1_s);
/// @endcode
```
The idea is that the users see right away how they can check the connection state.